### ISSUE TITLE
Vertex - incorrect discounted item rowTotal resulting in visually duplicate discount.

### ISSUE THREAD

**Issue by isaackim** [magento/magento2#27253]
### Preconditions (*)

1. Magento Enterprise 2.2.8
2. vertex/module-tax: 3.1.0
3. Vertex is set up and enabled

### Steps to reproduce (*)

1. Create a cart price rule
2. Add product(s) to cart
3. On the cart page apply discount code
4. Go to checkout page
5. Refresh the checkout page twice and observe the item rowTotal and subtotal values
6. Go back to cart and observe the item rowTotal and subtotal values

### Expected result (*)

1. ![A233BE5A-8596-4FE8-8175-AD161DF9B144](https://user-images.githubusercontent.com/1356575/76468789-0b190280-63aa-11ea-8595-e35a65c1a9cd.png)
2. ![BD4D6C07-605B-422D-B5E1-E9C1D824CF22](https://user-images.githubusercontent.com/1356575/76468788-0a806c00-63aa-11ea-95a7-aa12a1e18720.png)

### Actual result (*)

1. ![0C9B9DD3-A51C-4BAC-A869-4328D3758A3A](https://user-images.githubusercontent.com/1356575/76468790-0b190280-63aa-11ea-935f-550ce0011dd3.png)
2. ![FF557E70-F013-404A-A64A-6A92B26A8577](https://user-images.githubusercontent.com/1356575/76468786-09e7d580-63aa-11ea-9de6-37e4d2c5b85d.png)

### Additional information (*)
This issue only occurs when there is a need for tax calculation before user billing/shipping address is entered in (e.g. hence the refresh step on checkout page).
If billing or shipping is null, vertex creates a placeholder **taxDetailsItem (emptyDetailsTaxItem)** to avoid costly tax calculation until billing/shipping address is there.
The issue seems to coming from different **rowTotal** calculation formula for **emptyDetailsTaxItem** in `Vertex\Tax\Model\Calculator.php`

The **proposed solution** is:
```php
-        $rowTotal = ($quoteDetailsItem->getUnitPrice() * $quoteDetailsItem->getQuantity()) -
-            $quoteDetailsItem->getDiscountAmount();
+        $rowTotal = $quoteDetailsItem->getUnitPrice() * $quoteDetailsItem->getQuantity();
```

As soon as billing/shipping address is there, Vertex uses different formula to calculate **rowTotal** in `Vertex\Tax\Model\Api\Utility\PriceForTax.php`
```php
return (float) $this->calculationTool->round($item->getUnitPrice() * $item->getQuantity());
```

To support my **proposed solution**, here is how Magento calculates **rowTotal**:

`Magento\Tax\Model\Calculation\UnitBaseCalculator`
```php
->setRowTotal($price * $quantity)
```

`Magento\Tax\Model\Calculation\AbstractAggregateCalculator`
```php
$rowTotal = $price * $quantity;
```

**Comment by isaackim** [comment]
@navarr so you have visibility and can keep track

**Comment by navarr** [comment]
This issue is fixed in Vertex 3.4.0 which will be released with Magento 2.3.5
